diff --git a/conf.xml.tmpl b/conf.xml.tmpl
index d5329fb..1c67b9d 100644
--- a/conf.xml.tmpl
+++ b/conf.xml.tmpl
@@ -905,9 +905,9 @@
             <module uri="http://exist-db.org/xquery/cssparser"          class="org.exist.xquery.modules.cssparser.CSSParserModule"/>
             <module uri="http://exist-db.org/xquery/backups"            class="org.exist.backup.xquery.BackupModule" />
             <module uri="http://exist-db.org/xquery/cache"              class="org.exist.xquery.modules.cache.CacheModule" />
-            <module uri="http://exist-db.org/xquery/context"            class="org.exist.xquery.modules.context.ContextModule" />
+            <module uri="http://exist-db.org/xquery/context"            class="org.exist.xquery.modules.context.ContextModule" / -->
             <module uri="http://exist-db.org/xquery/cqlparser"          class="org.exist.xquery.modules.cqlparser.CQLParserModule" />
-            <module uri="http://exist-db.org/xquery/exi"                class="org.exist.xquery.modules.exi.ExiModule" />
+            <!-- module uri="http://exist-db.org/xquery/exi"                class="org.exist.xquery.modules.exi.ExiModule" />
             <module uri="http://exist-db.org/xquery/jndi"               class="org.exist.xquery.modules.jndi.JNDIModule" />
             <module uri="http://exist-db.org/xquery/jfreechart"         class="org.exist.xquery.modules.jfreechart.JFreeChartModule" />
             <module uri="http://exist-db.org/xquery/memcached"          class="org.exist.xquery.modules.memcached.MemcachedModule" />
@@ -916,7 +916,7 @@
             <module uri="http://exist-db.org/xquery/versioning/svn"     class="org.exist.versioning.svn.xquery.SVNModule" />
             <module uri="http://exist-db.org/xquery/xmpp"               class="org.exist.xquery.modules.xmpp.XMPPModule" />
             -->
-
+            <module uri="http://exist-db.org/xquery/exgit" class="org.exist.xquery.modules.exgit.Exgit"/>
             <!--
                 exiftool:
                 - perl-path - file system path to the perl executable

diff --git a/extensions/build.properties b/extensions/build.properties
index fe679a9..ef8be4d 100644
--- a/extensions/build.properties
+++ b/extensions/build.properties
@@ -96,7 +96,7 @@ include.module.context = false
 include.module.counter = true

 # Contextual Query Parser (CQL) module
-include.module.cqlparser = false
+include.module.cqlparser = true

 # Date/DateTime/Time utility functions module
 include.module.datetime = true
diff --git a/extensions/modules/cqlparser/ivy.xml b/extensions/modules/cqlparser/ivy.xml
index cee8c67..20d3abf 100644
--- a/extensions/modules/cqlparser/ivy.xml
+++ b/extensions/modules/cqlparser/ivy.xml
@@ -22,7 +22,7 @@
     <info organisation="org.exist" module="cqlparser"/>

     <dependencies>
-        <dependency org="org.z3950.zing" name="cql-java" rev="1.10" conf="*->*,!sources,!javadoc"/>
+        <dependency org="org.z3950.zing" name="cql-java" rev="1.13" conf="*->*,!sources,!javadoc"/>
     </dependencies>

 </ivy-module>
diff --git a/log4j2.xml b/log4j2.xml
index 5238d11..8e0d45b 100644
--- a/log4j2.xml
+++ b/log4j2.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Configuration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
     <Properties>
-        <Property name="logs">${log4j:configParentLocation}/webapp/WEB-INF/logs</Property>
+        <Property name="logs">$${env:EXIST_HOME}/webapp/WEB-INF/logs</Property>
         <Property name="rollover.max.size">10MB</Property>
         <Property name="rollover.max">14</Property>
         <Property name="rollover.file.pattern">%d{yyyyMMddHHmmss}</Property>
diff --git a/webapp/WEB-INF/web.xml.tmpl b/webapp/WEB-INF/web.xml.tmpl
index 898ead5..06b9951 100644
--- a/webapp/WEB-INF/web.xml.tmpl
+++ b/webapp/WEB-INF/web.xml.tmpl
@@ -97,7 +97,7 @@
 		-->
         <init-param>
             <param-name>hidden</param-name>
-            <param-value>false</param-value>
+            <param-value>true</param-value>
         </init-param>
 
         <load-on-startup>2</load-on-startup>
